pkg/cloudprovider/providers/aws/aws_metrics.go

```diff
@@ -53,8 +75,18 @@ func recordAWSThrottlesMetric(operation string) {
 	awsAPIThrottlesMetric.With(prometheus.Labels{"operation_name": operation}).Inc()
 }
 
+func recordAWSAttachDetachMetric(actionName string, node string, volume string, timeTaken float64, err error) {
+	if err != nil {
+		awsAttachDetachErrorMetric.With(prometheus.Labels{"operation": actionName, "node": node, "volume": volume}).Inc()
+	} else {
+		awsAttachDetachMetric.With(prometheus.Labels{"operation": actionName, "node": node, "volume": volume}).Observe(timeTaken)
```

Comment:
For successful attach or detach operations - adding volume and node name here will cause an explosion of time series metrics. Each metric with a combination of label and values is considered unique. So in a multitenant cluster with 4000 volumes on 200 nodes, this will cause creation of huge number of metric time series row.

Reply:
let's ditch it.

But how do we clean up errors? for other metrics we can look at state and implement Collector so that stale metrics go away.